### ISSUE TITLE
test_ffi_keygen_json_pair_dsa_elg fails occasionally.

### ISSUE THREAD

**Issue by ni4** [riboseinc/rnp#881]
## Description
[  FAILED  ] test_ffi_keygen_json_pair_dsa_elg

From https://github.com/riboseinc/rnp/pull/874#issuecomment-512902720:

It looks like it expects the key bits to be 1536 but it gets 1528. We may want to look more closely at that.
It seems to be a rare case when calculated Y value is 1 byte less then expected, however still need to reproduce it.

**Comment by ni4** [comment]
Here is some output from the custom test which generates ElGamal 1032-bit key material:
```
2: Iteration 29: 1026 bits
2: Iteration 30: 1032 bits
2: Iteration 31: 1029 bits
2: Iteration 32: 1032 bits
2: Iteration 33: 1032 bits
2: Iteration 34: 1032 bits
2: Iteration 35: 1032 bits
2: Iteration 36: 1031 bits
2: Iteration 37: 1029 bits
2: Iteration 38: 1031 bits
2: Iteration 39: 1032 bits
2: Iteration 40: 1023 bits
2: [  ERROR   ] --- (bits > 1024) && (bits <= 1032)
2: [   LINE   ] --- /Users/Ribose/Projects/rnp/src/tests/cipher.cpp:768: error: Failure!
2: [  FAILED  ] elgamal_generate_bits
2: [==========] 1 test(s) run.
2: [  PASSED  ] 0 test(s).
2: [  FAILED  ] 1 test(s), listed below:
2: [  FAILED  ] elgamal_generate_bits
```
So, it may fall as low as 1023 bits, and even less since key size is size of the `y = g ^ x mod q`, where x is chosen randomly.

**Comment by ronaldtse** [comment]
> The solution I see is to generate key material again if the generated key byte size is less then requested (however, not sure whether this is required, since x size is as requested).

Agree!

**Comment by ni4** [comment]
Fixed in PR #887
